UnitsNet/CustomCode/Wrappers/Pressure.Wrapper.cs

```diff
@@ -0,0 +1,129 @@
+        }
+
+        /// <summary>
+        ///     The public repersentation of the measured reference this quantity was constructed with.
+        /// </summary>
+        public PressureReference Reference => _reference.GetValueOrDefault(BaseReference);
+
+        /// <summary>
+        ///     The measured reference this quantity was constructed with.
+        /// </summary>
+        private readonly PressureReference? _reference;
+
+        /// <summary>
+        /// List property of reference options: Gauge, Absolute, and Vacuum
+        /// </summary>
+        public static List<PressureReference> References { get; } = Enum.GetValues(typeof(PressureReference)).Cast<PressureReference>().Except(new[] { PressureReference.Undefined }).ToList();
+
+        /// <summary>
+        ///     The base reference representation of this quantity for the numeric value stored internally. All conversions go via this value.
+        /// </summary>
+        public static PressureReference BaseReference { get; } = PressureReference.Absolute;
+
+        private Pressure Pressure { get; }
+
+        /// <summary>
+        ///     Get Gauge Pressure.
+        ///     It refers pressure level above Reference Pressure.
+        /// </summary>
+        public Pressure Gauge => As(PressureReference.Gauge);
+
+        /// <summary>
+        ///     Get Absolute Pressure.
+        ///     It is zero-referenced pressure to the perfect vacuum.
+        /// </summary>
+        public Pressure Absolute => As(PressureReference.Absolute);
+
+        /// <summary>
+        ///     Get Vacuum Pressure.
+        ///     It is a negative Gauge Pressure when Absolute Pressure is below Reference Pressure.
+        /// </summary>
+        public Pressure Vacuum => As(PressureReference.Vacuum);
+
+
+        private Pressure As(PressureReference reference)
+        {
+            var converted = AsBaseNumericType(reference);
+
+            return new Pressure(converted, Pressure.Unit);
+        }
+
+        private double AsBaseNumericType(PressureReference reference)
+        {
+            var baseReferenceValue = AsBaseReference();
+
+            if (Reference == reference)
+                return Pressure.Value;
+
+            var negatingValue = Reference == PressureReference.Vacuum ? -1 : 1;
+
+            switch (reference)
+            {
+                case PressureReference.Absolute: return baseReferenceValue;
+                case PressureReference.Gauge: return baseReferenceValue - ReferencedPressure.ToUnit(Pressure.Unit).Value;
+                case PressureReference.Vacuum: return ReferencedPressure.ToUnit(Pressure.Unit).Value - negatingValue * baseReferenceValue;
+                default:
+                    throw new NotImplementedException($"Can not convert {Reference} to {reference}.");
+            }
+        }
+
+        private double AsBaseReference()
+        {
+            switch (Reference)
+            {
+                case PressureReference.Absolute:
```

Comment:
I would suggest `{}` braces around multi-line statements inside switch cases. Single line cases don't need it.
You can also omit `else` when the `if` throws an exception.